src/app_config.rs

```diff
@@ -0,0 +1,52 @@
+use serde::Deserialize;
+use std::{collections::HashMap, fs, path::PathBuf};
+
+pub(crate) const CONFIG_FILE_NAME: &str = "cargo-generate";
+
+#[derive(Deserialize)]
+pub(crate) struct AppConfig {
+    pub favorites: HashMap<String, FavoriteConfig>,
+}
+
+#[derive(Deserialize)]
+pub(crate) struct FavoriteConfig {
+    pub description: Option<String>,
+    pub git: Option<String>,
+    pub branch: Option<String>,
+}
+
+pub(crate) fn edit_config(path: Option<PathBuf>) -> anyhow::Result<()> {
+    if let Some(path) = path {
+        println!(
+            r#"Attempting to edit config at "{}""#,
+            path.as_path().display().to_string()
+        );
+        if !path.exists() {
+            std::fs::File::create(&path)?;
+        }
+        open::that(path)?;
```

Comment:
I'd prefer to only present the user a message here to along the lines of

`println!("to edit your config: $EDITOR {}", path);`

That way we not need to pull in the `open` crate as a dependency, and keep it lean. Since the target user group is a shell user, we should keep things simple, that that would be directing the user `what to do` and `where to do it`

Reply:
The reason I used the `open` crate was actually to get it working on both windows and linux (and I've tested it)

I have no problem getting rid of it again - but that also means the `--edit-config` should be removed.
Then the question is: when/under what option, do we show where the config file is located??

Reply:
What about the [`edit crate`](https://docs.rs/edit/0.1.2/edit/)? 
Would also allow to fetch the input the user made directly inline here as return.

Reply:
I solved it by entirely removing the --edit-config option, then adding a better error description if/when users try to use the favorite feature with either no config or missing favorites.

So, better error descriptions and getting rid of the dependency.

If at some time we'd like the --edit-config feature, we should probably create a new feature request :-)